### PR TITLE
add tutorial document

### PR DESCRIPTION
**Pull request by ftian1**
## Type of Change

documentation change

## Description

This PR is used to add tutorial document for NeuralChat

### CODE REVIEW

**Comment by ftian1** [comment]
This is an initial version. some contents need to add. 

@XinyuYe-Intel  The OptimizationConfig definition and usage are not clean enough. pls follow this doc to revise API

**Comment by hshen14** [comment]
@ftian1  you can create the PR against master.

File: neural_chat/tutorial.md [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+| port                      |                          | 8000                                    |
+| audio                     | audio_output             | TRUE                                    |
+|                           | language                 | "english"                               |
+| retrieval                 | retrieval_type           | "dense"                                 |
+|                           | retrieval_document_path  | ../../assets/docs/                      |
+| caching                   | cache_chat_config_file   | ../../plugins/caching/cache_config.yaml |
+|                           | cache_embedding_model_dir| hkunlp/instructor-large                 |
+| model_name                |                          | meta-llama/Llama-2-7b-chat-hf           |
+
+## Practice on different Platforms
```
Comment (hshen14) [inline neural_chat/tutorial.md:41]: Deploy on Different Platforms

File: neural_chat/tutorial.md [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+chatbot = build_chatbot(config)
+
+## use chatbot to do prediction
+response = chatbot.predict("Tell me about Intel Xeon Scalable Processors.")
+```
+
+This way converts the underneath LLM model into `bfloat16` on the fly and fully leverage `Intel(R) AMX` technology's power to speed up the inference of the whole working flow.
+
+### Intel XEON Client Platforms
```
Comment (hshen14) [inline neural_chat/tutorial.md:64]: Remove this section first. And there is no Intel XEON Client

**Comment by hshen14** [comment]
Shall we create jupyter notebook (using docker file to simplify the setup)?

**Comment by ftian1** [comment]
> Shall we create jupyter notebook (using docker file to simplify the setup)?

the problem is the code is quite simple... and has been pasted in the tutorial doc. creating a new jupyter notebook looks duplicated.

**Comment by hshen14** [comment]
> > Shall we create jupyter notebook (using docker file to simplify the setup)?
> 
> the problem is the code is quite simple... and has been pasted in the tutorial doc. creating a new jupyter notebook looks duplicated.

Env setup is not that simple. we'll anyway need notebook esp. Gaudi2/client. so we can create notebooks in parallel. My understanding is for tutorials, we can explain more details from API level, but not necessarily deployment level.

**Comment by XinyuYe-Intel** [comment]
> This is an initial version. some contents need to add.
> 
> @XinyuYe-Intel The OptimizationConfig definition and usage are not clean enough. pls follow this doc to revise API

Current implementation is for the cases like combining AMP and weight only quantization, use the way in the doc won't be able to realize such cases.

**Comment by ftian1** [comment]
> > This is an initial version. some contents need to add.
> > @XinyuYe-Intel The OptimizationConfig definition and usage are not clean enough. pls follow this doc to revise API
> 
> Current implementation is for the cases like combining AMP and weight only quantization, use the way in the doc won't be able to realize such cases.

no, AMPConfig is purely for bfloat16 or float16 support. if user wants to do weight only quant and specify some ops to be float16, the WeightOnlyConfig should contain such parameter. The idea is each config means a quantization scheme on the model. it's self-contained and self-described. not need to combine with other config.

**Pull request closed without merging**